action/transfer.go

```diff
@@ -100,6 +108,27 @@ func (tsf *Transfer) Proto() *iotextypes.Transfer {
 	return act
 }
 
+// IsExternal returns true if the transfer has a valid external chain ID
+func (tsf *Transfer) IsExternal() bool {
+	return tsf.externChainID == config.ExternChainID()
+}
+
+// RawHash returns the tx's raw hash to be signed
+func (tsf *Transfer) RawHash() hash.Hash256 {
+	if !tsf.IsExternal() {
+		panic("should not call RawHash() on non-external transfer")
+	}
+	return tsf.rawHash
+}
+
+// Hash returns the signed hash
+func (tsf *Transfer) Hash() hash.Hash256 {
+	if !tsf.IsExternal() {
```

Comment:
This is risky. Hash() becomes only available for rlp transfers. non rlp-tsf causes panic

Reply:
here's the reason why i think panic() is better than return an error:

the ONLY legit source of RLP tx coming into our chain is from the wrapper proxy service, they come in as raw bytes, and gets processed by `sealEnvelope.LoadProto()`, where it computes the raw hash and hash, and saved in the RLP-transfer/execution

any normal tx (generated from ioctl, ioPay desktop, ioPay mobile, etc.) must be non-RLP, and RawHash(), Hash() must NOT be called on these normal tx. If it happens, means something is seriously wrong

since this func returns `hash.Hash256`, if don't panic, would need to return `hash.Hash256{}`, which is an all-0 hash, and no error is reported

so i would rather panic() to catch the unintended call, instead of let it silently pass

the same reason for this panic in `action.Sign`:
```
if act.payload.IsExternal() {
		panic("not allowed to sign external transaction")
	}
```